src/app/profile/payment-methods/payment-method/payment-method.component.js

```diff
@@ -0,0 +1,118 @@
+import angular from 'angular';
+import template from './payment-method.tpl';
+import displayAddressComponent from 'common/components/display-address/display-address.component';
+import recurringGiftsComponent from '../recurring-gifts/recurring-gifts.component';
+import paymentMethodFormModal from 'common/components/paymentMethods/paymentMethodForm/paymentMethodForm.modal.component';
+import deletePaymentMethodModal from 'common/components/paymentMethods/deletePaymentMethod/deletePaymentMethod.modal.component.js';
+import giveModalWindowTemplate from 'common/templates/giveModalWindow.tpl';
+import profileService from 'common/services/api/profile.service';
+
+let componentName = 'paymentMethod';
+
+class PaymentMethodController{
+
+  /* @ngInject */
+  constructor(envService, $uibModal, profileService){
+    this.isCollapsed = true;
+    this.$uibModal = $uibModal;
+    this.profileService = profileService;
+    this.imgDomain = envService.read('imgDomain');
+    this.submissionError = {error: ''};
+    this.submitted = false;
+  }
+
+  $onInit(){
+    this.loadDonorDetails();
+  }
+
+  loadDonorDetails() {
+    this.profileService.getDonorDetails()
+      .subscribe((data) => {
+        this.mailingAddress = data.mailingAddress;
+      });
+  }
+
+  getExpiration(){
+    return `${this.model['expiry-month']}/${this.model['expiry-year']}`;
+  }
+
+  isCard(){
+    return this.model['card-number'] ? true : false;
+  }
+
+  editPaymentMethod() {
+    this.editPaymentMethodModal = this.$uibModal.open({
+      component: 'paymentMethodFormModal',
+      windowTemplateUrl: giveModalWindowTemplate.name,
+      resolve: {
+        paymentMethod: () => this.model,
+        mailingAddress: () => this.mailingAddress,
+        onSubmit: () => this.onSubmit,
+        submitted: () => this.submitted,
+        submissionError: () => this.submissionError,
+        profileService: () => this.profileService
+      }
+    });
+    this.editPaymentMethodModal.result.then(() => {
```

Comment:
I don't think you need these lines. If the modal has a result, that means it has already been closed.